parser.go

```diff
@@ -1017,8 +1017,10 @@ func (p *parser) getSchemaObjectCached(pkgPath, pkgName, typeName string) (*Sche
 	// see if we've already parsed this type
 	if knownObj, ok := p.KnownIDSchema[genSchemeaObjectID(pkgName, typeName)]; ok {
 		schemaObject = knownObj
-	} else {
+	} else if foundType, ok := p.KnownIDSchema[typeName]; ok {
```

Comment:
@bwoskow-ld without this logic the JsonPatch model fails for code refs with this error: `2021/06/08 14:52:14 schema does not reference valid name`

I have not been able to find a better way to handle this yet. If it's not found it should just parse the rep but for some reason it doesn't appear to be.

Reply:
Can you provide an example of how this impacts the actual schema? I'm having trouble telling this from just looking at the example. Is it the case that we have `$ref` objects which reference non-existent items?